### PR TITLE
100% coverage of c.fsw

### PR DESCRIPTION
**Pull request by Zain2050**
There was no template for cp_fs2_toggle, so I made one. 

Abdullah got 100% coverage of c.flw, but later it dropped to 80% (also for F extension flw). We both looked into the problem but couldn't figure out what really caused it. However, I fixed it by changing **fsd_val** to **fsd_val[31:0]** in **cp_fd_toggle.txt**. Now, both c.flw & c.fsw are at 100%.

### CODE REVIEW

**Comment by davidharrishmc** [comment]
Glad to see a fix.  We'd better check if cp_fd_toggle is used in any of the D tests - if so, they will need a 64-bit version.  @coreyqh could you review?

Also, likes 322-334 have a fair amount of repeated code.  Could you restructure this to share the code that is identical between the branches of the if?

**Comment by Zain2050** [comment]
It would be great if the HMC team keeps a close eye on compressed floating instructions, as we don't want to end up breaking something for the F or D extension.

**Comment by coreyqh** [comment]
It looks likes none of the D extension instructions have fd_toggle selected. Is there value to be gained by having this coverpoint? if so, I'll incorporate into all of the testplans. 

It is a good idea to replace the top row of your testplan with the one used in all other floating point testplans, for consistency's sake. There might be a bit of refactoring to do but it will make it easy to propagate bug fixes throughout the floating point tests. On that same note, if you find an issue with one of the floating point tests (such as missing or incorrect coverage), **send me a slack message outlining the issue and why it must change**. This way, I will make the same change across all of the testplans in one go. 

I can review compressed fp instructions, feel free to message me on slack if you need anything specific looked at. 

Just to be clear, you fixed the missing flw coverage in the 32F extension? 

Great work UET team!

**Comment by Zain2050** [comment]
@coreyqh Appreciate it! I'll reach you out on Slack if I have anything to discuss about floats. And it's great to hear that this fixed the missing coverage for flw.

Well, I don't think fd_toggle is a crucial coverpoint. It's totally up to you if you want to add it to your tests or not. I'll replace the top row of the testplan as it sounds like a good idea.

**Comment by UmerShahidengr** [comment]
Good idea @coreyqh ,and good work @Zain2050 replacing the top line of Zcf coverage and F.
I think it is good to go. Anybody can merge this one

**Review by UmerShahidengr — APPROVED** [review]

File: templates/RV32_cp_fd_toggle.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-    cp_fd_toggle : coverpoint unsigned'(ins.current.fd_val)  iff (ins.trap == 0 )  {
+    cp_fd_toggle : coverpoint unsigned'(ins.current.fd_val[31:0])  iff (ins.trap == 0 )  {
```
[the remark this answers is not in the document]
Comment (Zain2050) [reply templates/RV32_cp_fd_toggle.txt:1]: I didn't see your question earlier. fd_val is of 64 bits, where 32 bits are signed extended to 64 bits. Due to this difference in width, it wasn't being compared.
`1111111111111111111111111111111110100010000011101000110101001001`
XLEN is 32, so fd_val is probably FLEN as it is 64. This seems strange as the tests are for single precision floats, so FLEN should have been 32, perhaps there's some special reason behind it.

**Comment by davidharrishmc** [comment]
XLEN and FLEN come from the configuration file, not the tests.  RV32GC has XLEN=32 because it is an RV32 variant.  G implies D and F, so FLEN=64 because the datapath supports doubles.  When running float code on RV32GC, the 32-bit numbers are NaN-boxed into the 64-bit floating-point registers.  This is why you needed to look at the bottom 32 bits.

In RV64GC, XLEN=FLEN=64.

Our tests should work on any configuration, including hypothetical RV32IF, which would have only XLEN=FLEN=32.

**Comment by davidharrishmc** [comment]
I think you'll need to push changes to cvw config to have your new coverage show up in everyone's regression.

**Comment by UmerShahidengr** [comment]
Thank you for detailed reply @davidharrishmc ,  now it makes sense to do [31:0] with float value.

**Pull request merged**